### ISSUE TITLE
Style/RedundantSort crashes

### ISSUE THREAD

**Issue by ccutrer** [rubocop/rubocop#10118]
I'm getting some errors running RuboCop against some code. Details below.

--------

## Expected behavior

RuboCop doesn't give an error.

## Actual behavior

```
An error occurred while Style/RedundantSort cop was inspecting /Users/cody/src/canvas-lms/app/controllers/courses_controller.rb:531:27.
undefined method `method?' for #<RuboCop::AST::OrNode:0x00007fad63210e18>
Did you mean?  method
               methods
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/style/redundant_sort.rb:96:in `use_size_method_in_block?'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/style/redundant_sort.rb:83:in `on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:136:in `public_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:136:in `block (2 levels) in trigger_restricted_cops'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:135:in `block in trigger_restricted_cops'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:134:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:134:in `trigger_restricted_cops'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:70:in `on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:154:in `on_block'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:159:in `block in on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:156:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:156:in `each_with_index'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:156:in `on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:159:in `block in on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:156:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:156:in `each_with_index'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:156:in `on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:155:in `on_if'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_if'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:158:in `on_block'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:154:in `on_def'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_def'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:154:in `on_class'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_class'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/team.rb:155:in `investigate_partial'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/team.rb:83:in `investigate'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:309:in `inspect_file'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:253:in `block in do_inspection_loop'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:287:in `block in iterate_until_no_changes'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:280:in `loop'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:280:in `iterate_until_no_changes'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:249:in `do_inspection_loop'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:130:in `block in file_offenses'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:155:in `file_offense_cache'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:129:in `file_offenses'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:67:in `block in warm_cache'
/Users/cody/.gem/ruby/2.7.2/gems/parallel-1.21.0/lib/parallel.rb:515:in `call_with_index'
/Users/cody/.gem/ruby/2.7.2/gems/parallel-1.21.0/lib/parallel.rb:485:in `process_incoming_jobs'
/Users/cody/.gem/ruby/2.7.2/gems/parallel-1.21.0/lib/parallel.rb:465:in `block in worker'
/Users/cody/.gem/ruby/2.7.2/gems/parallel-1.21.0/lib/parallel.rb:456:in `fork'
/Users/cody/.gem/ruby/2.7.2/gems/parallel-1.21.0/lib/parallel.rb:456:in `worker'
/Users/cody/.gem/ruby/2.7.2/gems/parallel-1.21.0/lib/parallel.rb:447:in `block in create_workers'
/Users/cody/.gem/ruby/2.7.2/gems/parallel-1.21.0/lib/parallel.rb:446:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/parallel-1.21.0/lib/parallel.rb:446:in `each_with_index'
/Users/cody/.gem/ruby/2.7.2/gems/parallel-1.21.0/lib/parallel.rb:446:in `create_workers'
/Users/cody/.gem/ruby/2.7.2/gems/parallel-1.21.0/lib/parallel.rb:386:in `work_in_processes'
/Users/cody/.gem/ruby/2.7.2/gems/parallel-1.21.0/lib/parallel.rb:289:in `map'
/Users/cody/.gem/ruby/2.7.2/gems/parallel-1.21.0/lib/parallel.rb:238:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:67:in `warm_cache'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:46:in `run'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli/command.rb:11:in `run'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli/environment.rb:18:in `run'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli.rb:71:in `run_command'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli.rb:78:in `execute_runners'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli.rb:47:in `run'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/exe/rubocop:12:in `block in <top (required)>'
/Users/cody/.rubies/ruby-2.7.2/lib/ruby/2.7.0/benchmark.rb:308:in `realtime'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/exe/rubocop:12:in `<top (required)>'
/Users/cody/.gem/ruby/2.7.2/bin/rubocop:23:in `load'
/Users/cody/.gem/ruby/2.7.2/bin/rubocop:23:in `<main>'
Inspecting 1 file
Scanning /Users/cody/src/canvas-lms/app/controllers/courses_controller.rb
An error occurred while Style/RedundantSort cop was inspecting /Users/cody/src/canvas-lms/app/controllers/courses_controller.rb:531:27.
undefined method `method?' for #<RuboCop::AST::OrNode:0x00007fad63208768>
Did you mean?  method
               methods
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/style/redundant_sort.rb:96:in `use_size_method_in_block?'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/style/redundant_sort.rb:83:in `on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:136:in `public_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:136:in `block (2 levels) in trigger_restricted_cops'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:135:in `block in trigger_restricted_cops'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:134:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:134:in `trigger_restricted_cops'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:70:in `on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:154:in `on_block'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:159:in `block in on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:156:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:156:in `each_with_index'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:156:in `on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:159:in `block in on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:156:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:156:in `each_with_index'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:156:in `on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_send'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:155:in `on_if'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_if'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:158:in `on_block'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:154:in `on_def'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_def'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:154:in `on_class'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_class'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-ast-1.11.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/team.rb:155:in `investigate_partial'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cop/team.rb:83:in `investigate'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:309:in `inspect_file'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:253:in `block in do_inspection_loop'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:287:in `block in iterate_until_no_changes'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:280:in `loop'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:280:in `iterate_until_no_changes'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:249:in `do_inspection_loop'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:130:in `block in file_offenses'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:155:in `file_offense_cache'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:129:in `file_offenses'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:120:in `process_file'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:101:in `block in each_inspected_file'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:100:in `each'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:100:in `reduce'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:100:in `each_inspected_file'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:86:in `inspect_files'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/runner.rb:47:in `run'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli/command.rb:11:in `run'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli/environment.rb:18:in `run'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli.rb:71:in `run_command'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli.rb:78:in `execute_runners'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/lib/rubocop/cli.rb:47:in `run'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/exe/rubocop:12:in `block in <top (required)>'
/Users/cody/.rubies/ruby-2.7.2/lib/ruby/2.7.0/benchmark.rb:308:in `realtime'
/Users/cody/.gem/ruby/2.7.2/gems/rubocop-1.21.0/exe/rubocop:12:in `<top (required)>'
/Users/cody/.gem/ruby/2.7.2/bin/rubocop:23:in `load'
/Users/cody/.gem/ruby/2.7.2/bin/rubocop:23:in `<main>'
C

Offenses:

app/controllers/courses_controller.rb:528:23: C: [Correctable] Style/RedundantSort: Use min_by instead of sort_by...first.
      e = enrollments.sort_by { |e| e.state_with_date_sortable }.first
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense auto-correctable

1 error occurred:
An error occurred while Style/RedundantSort cop was inspecting /Users/cody/src/canvas-lms/app/controllers/courses_controller.rb:531:27.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.21.0 (using Parser 3.0.2.0, rubocop-ast 1.11.0, running on ruby 2.7.2 x86_64-darwin20)
Finished in 2.165997999996762 seconds
```

## Steps to reproduce the problem

`rubocop --only Style/RedundantSort -d -- app/controllers/courses_controller.rb`

The file in question is https://github.com/instructure/canvas-lms/blob/725711463b386ce54e91032518ffaa6c8b8d0188/app/controllers/courses_controller.rb

## RuboCop version

```
1.21.0 (using Parser 3.0.2.0, rubocop-ast 1.11.0, running on ruby 2.7.2 x86_64-darwin20)
  - rubocop-performance 1.11.5
  - rubocop-rails 2.12.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.5.0
```

**Comment by dvandersluis** [comment]
Here is a boiled down example of the failure:

```ruby
x.sort_by { |y| y.foo || bar }.last
```

Looking into it now.
